### PR TITLE
fix: routes ui for bridge flow

### LINKED ISSUE #19970: Routes UI for Bridge flow are broken
UI for routes on bridge flow is broken. The + button should not be displayed in `To` column because the user needs to select a network to bridge to. Also the selected network is not showing.

Current behavior:

https://github.com/status-im/status-mobile/assets/18485527/0a6228b8-a55c-458e-a78d-a98f24d81431

Expected behavior:

- The + button should not be displayed in `To` column 
- The selected network should be shown in `To` column

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #19970

### Summary

This PR fixes broken UI for routes on bridge flow (+ button should not be displayed in `To` column, also selected network is not showing). Also added new unit tests on relevant utils functions to cover these cases.

### Testing notes

Bridge flow seems broken and user can't confirm Bridge operation, this PR only fixes routes ui.

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

### Steps to test

- Open Status
- Login
- Go to wallet
- Go to account
- Tap on Bridge
- Select a network to bridge to
- Enter a valid amount
- Check routes UI are displayed correctly

#### Before

https://github.com/status-im/status-mobile/assets/18485527/9004400b-2faf-4d7b-afe6-4ac7fd277bf8

#### After

https://github.com/status-im/status-mobile/assets/18485527/2691e363-f1b0-46b4-80dc-b1c361d2a8d6

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/send/events.cljs [hunk 1]
```diff
@@ -286,15 +289,22 @@
                                                                       disabled-from-chain-ids}))
          sender-network-values (when token-available-networks-for-suggested-routes
                                  (send-utils/loading-network-amounts
-                                  token-available-networks-for-suggested-routes
+                                  (if (= transaction-type :bridge)
+                                    (filter #(not= bridge-to-chain-id %)
+                                            token-available-networks-for-suggested-routes)
+                                    token-available-networks-for-suggested-routes)
```
Comment (OmarBasem) [inline src/status_im/contexts/wallet/send/events.cljs]: I am not sure what does `token-available-networks-for-suggested-routes` mean, is it the available routes for the user's network preferences?

File: src/status_im/contexts/wallet/send/events.cljs [hunk 2]
```diff
@@ -56,13 +57,15 @@
                                            (send-utils/network-amounts from-network-values-for-ui
                                                                        disabled-from-chain-ids
                                                                        receiver-networks
+                                                                       transaction-type
```
Comment (OmarBasem) [inline src/status_im/contexts/wallet/send/events.cljs]: I see sometimes `transaction-type` and others `tx-type`

File: src/status_im/contexts/wallet/send/utils.cljs [hunk 3]
```diff
@@ -140,15 +144,18 @@
                                              :disabled)}
                           (and (not to?) (contains? disabled-set k))
                           (assoc :total-amount (money/bignumber "0")))))
+                 (filter
+                  #(or (and to?
+                            (or
+                             (= tx-type :bridge)
+                             (contains? receiver-networks-set (:chain-id %))))
+                       (not to?)))
```
Comment (OmarBasem) [inline src/status_im/contexts/wallet/send/utils.cljs]: What is `to?`

**Comment by OmarBasem** [comment]
PR description missing issue number, is there an open issue to be linked?

File: src/status_im/contexts/wallet/send/utils.cljs (hunk 3 above)
Comment (briansztamfater) [reply src/status_im/contexts/wallet/send/utils.cljs]: The logic differs a bit if we are calculating networks for the From column or the To column, so it is a flag that indicates whether if we are calculating things for receiver or sender.

File: src/status_im/contexts/wallet/send/events.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/status_im/contexts/wallet/send/events.cljs]: Are the possible networks where a route could be found for the selected token.

File: src/status_im/contexts/wallet/send/events.cljs (hunk 2 above)
Comment (briansztamfater) [reply src/status_im/contexts/wallet/send/events.cljs]: I'll refactor to just use `tx-type` 👍

**Comment by briansztamfater** [comment]
> PR description missing issue number, is there an open issue to be linked?

@OmarBasem I haven't found one, I found the bug while testing and started to work on it, I will create one issue edit the description.

EDIT: Created #19970 and linked to it in the description

File: src/status_im/contexts/wallet/send/utils.cljs (hunk 3 above)
Comment (briansztamfater) [reply src/status_im/contexts/wallet/send/utils.cljs]: Refactored this on other PR. Check https://github.com/status-im/status-mobile/pull/19674#discussion_r1597723493

File: src/status_im/contexts/wallet/send/events.cljs (hunk 2 above)
Comment (briansztamfater) [reply src/status_im/contexts/wallet/send/events.cljs]: Updated!

File: src/status_im/contexts/wallet/send/utils.cljs (hunk 3 above)
Comment (OmarBasem) [reply src/status_im/contexts/wallet/send/utils.cljs]: I think you can refactor the relevant changes here too, or they may overwrite the other changes (?)

**Review by OmarBasem — APPROVED** [review]

File: src/status_im/contexts/wallet/send/utils.cljs (hunk 3 above)
Comment (briansztamfater) [reply src/status_im/contexts/wallet/send/utils.cljs]: They will conflict when I merge one of both PRs so I prefer to so refactor them once and rebase

**Review by J-Son89 — APPROVED** [review]

**Comment by qoqobolo** [comment]
Testing of wallet PRs is blocked until the problem with networks/balances is resolved https://github.com/status-im/status-mobile/issues/20011

**Comment by briansztamfater** [comment]
> Testing of wallet PRs is blocked until the problem with networks/balances is resolved #20011

@qoqobolo are balances not fetching for non-testnet networks? AFAIK, bridges are not supported on Sepolia (at least as of now), so this should be tested on mainnet.

**Comment by briansztamfater** [comment]
BTW, I need to solve conflicts before testing

**Comment by qoqobolo** [comment]
> @qoqobolo are balances not fetching for non-testnet networks? AFAIK, bridges are not supported on Sepolia (at least as of now), so this should be tested on mainnet.

Hey @briansztamfater, tbh it is not completely clear what and where exactly the problem is: those banners with `The providers for Arbitrum, Mainnet, Optimism are down` sometimes appear both on testnet and mainnet. 
Yesterday it was impossible to test this PR, today it looks better, so hopefully we can test it soon.

It might also be better to [test and update go version](https://github.com/status-im/status-mobile/pull/20061) which includes changes to the router first and then test this PR, WDYT?

**Comment by briansztamfater** [comment]
No problem @qoqobolo, I finished with the rebase and solving conflicts, so now is ready to test whenever you feel the conditions are met 👍 .

**Comment by qoqobolo** [comment]
Thanks for your work and patience @briansztamfater!
PR can be merged.

**Pull request merged**